AbsEngine/ECS/ComponentSystem.cs

```diff
@@ -2,58 +2,60 @@
 
 public abstract class ComponentSystem<T> : System where T : Component
 {
-    protected virtual Func<T, bool>? Predicate => null;
-
-    protected virtual bool UseParallel => false;
-
     protected virtual int MaxIterationsPerFrame => int.MaxValue;
-    protected virtual float? FixedTimeStep => null;
 
-    float lastTickTime = 0;
+    private int _componentsToSkip = 0;
 
     protected ComponentSystem(Scene scene) : base(scene)
     {
     }
 
     public virtual void OnInitialiseTick(float deltaTime) { }
 
-    public override void Tick(float deltaTime)
+    public override void OnTick(float deltaTime)
     {
-        if (FixedTimeStep != null)
+        List<T> comps;
+        int count = 0;
+        using (Profiler.BeginEvent("Gather Components"))
         {
-            if (lastTickTime >= FixedTimeStep.Value)
+            IEnumerable<T> initQuery;
+            using (Profiler.BeginEvent($"Scene.EntityManager.GetComponents<{typeof(T)}>"))
+                initQuery = Scene.EntityManager.GetComponents<T>(out count);
```

Comment:
For future: We need to optimise the way we get the list of components out of the system